tests/config_test.py

```diff
@@ -95,3 +101,92 @@ def test_load_minigraph_with_disabled_telemetry(self, get_cmd_module, setup_sing
     def teardown_class(cls):
         os.environ['UTILITIES_UNIT_TESTING'] = "0"
         print("TEARDOWN")
+
+
+class TestConfigQos(object):
+    @classmethod
+    def setup_class(cls):
+        print("SETUP")
+        os.environ['UTILITIES_UNIT_TESTING'] = "2"
+        import config.main
+        imp.reload(config.main)
+
+    def test_qos_reload_single(self, get_cmd_module, setup_single_broadcom_asic):
+        (config, show) = get_cmd_module
+        runner = CliRunner()
+        output_file = os.path.join(os.sep, "tmp", "qos_config_output.json")
+        print("Saving output in {}".format(output_file))
+        try:
+            os.remove(output_file)
+        except OSError:
+            pass
+        json_data = '{"DEVICE_METADATA": {"localhost": {}}}'
+        result = runner.invoke(
+            config.config.commands["qos"],
+            ["reload", "--dry_run", output_file, "--json-data", json_data]
+        )
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+
+        cwd = os.path.dirname(os.path.realpath(__file__))
+        expected_result = os.path.join(
+            cwd, "qos_config_input", "config_qos.json"
+        )
+        assert filecmp.cmp(output_file, expected_result, shallow=False)
```

Comment:
Please validate the buffer template as well for single and masic . That also gets generated in the 'qos reload' command.

Reply:
The main purpose of this test is to validate qos configs are generated for all ASICs on a mulit ASIC platform. 'sonic-cfgen' tests have the tests for qos and buffer templates.
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/tests/test_j2files.py
" test_qos_dell6100_render_template, test_buffers_dell6100_render_template"